client_test/directfund_test.go

```diff
@@ -58,7 +59,7 @@ func TestWhenObjectiveIsRejected(t *testing.T) {
 	outcome := testdata.Outcomes.Create(alice.Address(), bob.Address(), ledgerChannelDeposit, ledgerChannelDeposit)
 	response := clientA.CreateLedgerChannel(bob.Address(), 0, outcome)
 
-	waitTimeForCompletedObjectiveIds(t, &clientA, time.Second, response.Id)
+	<-clientA.WaitForObjectivesToComplete(context.Background(), response.Id)
```

Comment:
Outside of the scope of this PR but would it make sense for any of the other API functions to take in a context?